src/utils/PushQueue.ts

```diff
@@ -247,16 +236,23 @@ export default class PushQueue<T> {
     async _cleanup() {
         this.finished = true
         const { error } = this
-        for (const p of this.nextQueue) {
+        const queue = this.nextQueue
+        this.error = undefined
+        this.nextQueue = []
+        this.pending = 0
+        this.buffer.length = 0
+        while (queue.length) {
+            const p = queue.shift()
+            if (!p) { continue }
```

Comment:
Compare explicitly to `undefined`, so that the queue can contain falsy values.

Reply:
Good thought, though the "queue" here doesn't contain values, it's a queue of promises generated by calls to `next()` while the current async `next` is inflight i.e. should never contain anything but `Defer` promises. The `this.buffer` is what contains the buffered values, though given this class is called `PushQueue` I can see the confusion. I'll rename the `queue` variable to `nextQueue`